spec/workers/sync_permissions_worker_spec.rb

```diff
@@ -7,10 +7,39 @@
     is_expected.to be_processed_in :user
   end
 
-  it "should sync user permissions" do
+  it "should call to sync user permissions" do
     user = create(:user)
     expect(User).to receive(:find_by_id).with(user.id).and_return(user)
     expect(user).to receive(:update_repo_permissions)
     subject.perform(user.id)
   end
+
+  context "with recorded data" do
+    before do
+      freeze_time
+    end
+
+    it "should sync user permissions and repositories" do
+      VCR.use_cassette("github/user_sync") do
+        user = create(:user)
+        user.identities.update_all(token: "TEST_TOKEN")
+
+        expect(user.repository_permissions.count).to be 0
+        expect(Repository.count).to be 0
+        expect(RepositoryDownloadWorker.jobs.size).to be 0
+
+        expect(User).to receive(:find_by_id).with(user.id).and_return(user)
+        subject.perform(user.id)
+
+        # expect permissions were added
+        expect(user.repository_permissions.count).to be 109
+        # expect repositories for those permissions to have been added
+        expect(Repository.count).to be 109
+        # expect work to be queued up to sync data for those repositories
+        expect(RepositoryDownloadWorker.jobs.size).to be 218
```

Comment:
Do these actually get queued up/run in the test env or has all that been stubbed?

Reply:
They are only queued, not run.